ldms/python/ldms.pyx

```diff
@@ -1947,7 +1947,8 @@ cdef class Xprt(object):
                 this function will just be blocked indefinitely.
         """
         if self._conn_cb:
-            raise RuntimeError("The callback has been supplied to `connect()`. "
+            raise TypeError("Bad `Xprt.recv()` call. "
```

Comment:
@narategithub I think this should be a ValueError, from the Python3 docs:

```
Raised when an operation or function receives an argument that has the right type 
but an inappropriate value...
```

Reply:
Done. I was reluctant between `TypeError` and `ValueError`. Neither of them are exactly fit because this function does not receive an argument, but I think they are the closest to this error: using synchronous recv() on an asynchronous transport.